server/engine/engine.go

```diff
@@ -164,6 +181,8 @@ func (engine *Engine) waitBeforeEnding() {
 		case <-engine.context.Done():
 		}
 	}
+	// Add DeploymentMetric to Database
+	engine.SetFinalStatus()
```

Comment:
I am thinking this call should go before the sleep/wait, since it is at that point that the deployment is finished.  This is adding 15 min to the deployment time.